File: src/main/java/site/gachontable/gachontablebe/domain/admin/usecase/UpdateStatus.java
package site.gachontable.gachontablebe.domain.admin.usecase;

import site.gachontable.gachontablebe.domain.admin.presentation.dto.request.UpdateStatusRequest;
import site.gachontable.gachontablebe.domain.auth.domain.AuthDetails;
import site.gachontable.gachontablebe.domain.shared.dto.response.RegisterResponse;

public interface UpdateStatus {
    RegisterResponse executeForOpenStatus(AuthDetails authDetails, UpdateStatusRequest request);

    RegisterResponse executeForWaitingStatus(AuthDetails authDetails, UpdateStatusRequest request);
}

File: src/main/java/site/gachontable/domain/admin/port/in/UpdateStatus.kt
package site.gachontable.domain.admin.port.in;

import site.gachontable.presentation.admin.dto.request.UpdateStatusRequest;
import site.gachontable.infra.security.principal.AuthDetails;
import site.gachontable.presentation.shared.dto.response.RegisterResponse;

public interface UpdateStatus {

    RegisterResponse executeForOpenStatus(AuthDetails authDetails, UpdateStatusRequest request);

    RegisterResponse executeForWaitingStatus(AuthDetails authDetails, UpdateStatusRequest request);
}
